lib/transactionsFilter/TransactionHashesCache.js

```diff
@@ -3,12 +3,22 @@ const {
   util: { buffer: BufferUtils },
 } = require('@dashevo/dashcore-lib');
 
+const BLOCKS_TO_STAY_IN_INSTANT_LOCK_CACHE = 10;
+
 class TransactionHashesCache {
   constructor() {
     this.transactions = [];
     this.merkleBlocks = [];
     this.blocks = [];
     this.cacheSize = 10;
+
+    // Instant lock cache
+    this.transactionHashesMap = Object.create(null);
```

Comment:
Does it make sense to keep array and map? It almost the same as what you had in two different classes.
Maybe then it's better to have one map with `transaction`, `blockHeight` and `isRetrieved`? Then you will speed up many operations on `this.trasactions`